### PR TITLE
Update db to allow the use of several primary keys

### PR DESCRIPTION
**Pull request by pportelette**

### CODE REVIEW

File: src/Pachyderm/Db.php [hunk 1]
```diff
@@ -189,7 +200,19 @@ public static function update($table, array $content, $where) {
      * @return false|true Return true if success, false otherwise
      */
     public static function delete($table, $key, $value) {
-        return self::query('DELETE FROM `' . $table . '` WHERE `' . $key . '`="' . Db::escape($value) . '"');
+        if (!is_array($key) && !is_array($value)) {
+            return self::query('DELETE FROM `' . $table . '` WHERE `' . $key . '`="' . Db::escape($value) . '"');
+        } else {
```
Comment (mermetbt) [inline src/Pachyderm/Db.php]: What's the purpose of having `else` here?

File: src/Pachyderm/Db.php [hunk 2]
```diff
@@ -132,7 +132,18 @@ public static function findAll($table, $where = NULL, $order = NULL, $offset = 0
      * @return false|array Return element if success, false otherwise
      */
     public static function findOne($table, $key, $value) {
-        $result = Db::query('SELECT * FROM `' . $table . '` WHERE `' . $key . '`="' . Db::escape($value) . '"');
+        if (!is_array($key) && !is_array($value)) {
+            $result = Db::query('SELECT * FROM `' . $table . '` WHERE `' . $key . '`="' . Db::escape($value) . '"');
+        } else {
+            $query = 'SELECT * FROM `' . $table . '` WHERE ';
+            for ($i = 0; $i < count($key); $i++) {
```
Comment (mermetbt) [inline src/Pachyderm/Db.php]: `count` is executed several time in here, that's loosing performance.

File: src/Pachyderm/Db.php [hunk 3]
```diff
@@ -132,7 +132,18 @@ public static function findAll($table, $where = NULL, $order = NULL, $offset = 0
      * @return false|array Return element if success, false otherwise
      */
     public static function findOne($table, $key, $value) {
-        $result = Db::query('SELECT * FROM `' . $table . '` WHERE `' . $key . '`="' . Db::escape($value) . '"');
+        if (!is_array($key) && !is_array($value)) {
+            $result = Db::query('SELECT * FROM `' . $table . '` WHERE `' . $key . '`="' . Db::escape($value) . '"');
```
Comment (mermetbt) [inline src/Pachyderm/Db.php]: Code `partial` duplication here.

**Review by mermetbt — COMMENTED** [review]
Review done

File: src/Pachyderm/Db.php (hunk 3 above)
Comment (pportelette) [reply src/Pachyderm/Db.php]: You mean with the delete function ?

**Pull request merged**